dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -535,25 +483,54 @@ private void handleTrackedEntityAttributeValues( Session session, TrackerPreheat
                     attributeValue.getValue() );
             }
         }
-
-        assignedFileResources.forEach( fr -> assignFileResource( session, preheat, fr ) );
-        unassignedFileResources.forEach( fr -> unassignFileResource( session, preheat, fr ) );
     }
 
-    private void assignFileResource( Session session, TrackerPreheat preheat, String fr )
+    private void handleDataValues( Session session, TrackerPreheat preheat, Set<DataValue> payloadDataValues,
+        ProgramStageInstance psi )
     {
-        FileResource fileResource = preheat.get( TrackerIdScheme.UID, FileResource.class, fr );
+        Map<String, EventDataValue> dataValueDBMap = psi
+            .getEventDataValues()
+            .stream()
+            .collect( Collectors.toMap( dv -> dv.getDataElement(),
+                dv -> dv ) );
 
-        if ( fileResource == null )
+        for ( DataValue dv : payloadDataValues )
         {
-            return;
+            DataElement dateElement = preheat.get( TrackerIdScheme.UID, DataElement.class, dv.getDataElement() );
+
+            checkNotNull( dateElement,
+                "Data element should never be NULL here if validation is enforced before commit." );
+
+            EventDataValue eventDataValue = dataValueDBMap.getOrDefault( dv.getDataElement(), new EventDataValue() );
+
+            eventDataValue.setDataElement( dateElement.getUid() );
+            eventDataValue.setValue( dv.getValue() );
+            eventDataValue.setStoredBy( dv.getStoredBy() );
+
+            if ( StringUtils.isEmpty( eventDataValue.getValue() ) )
+            {
+                unassignFileResource( session, preheat, dataValueDBMap.get( dv.getDataElement() ).getValue() );
```

Comment:
I'm a bit puzzled regarding this code. The current tracker, as far as I know, does not handle file resources associated to Event Data Values. Can someone confirm @mortenoh @stian-sandvold